shared/src/commonMain/kotlin/com/example/tasky/manager/SessionManager.kt

```diff
@@ -95,4 +95,16 @@ object SessionManager {
             println("cannot get name: $e")
             null
         }
+
+    suspend fun getUserId(): String? =
+        try {
+            val loginResponse = loadLoginResponse()
+
+            loginResponse?.userId
+        } catch (e: Exception) {
+            if (e is CancellationException) throw e
+
+            println("cannot get userId: $e")
+            null
+        }
```

Comment:
Here, you have a crucial situation in your app where you don’t have user session information when you should have it 99% of the time when you call this function. Make sure to handle the cases where you receive `null` here and then reset your app state because the session is no longer valid.